// @vidstack/react/dist/server.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AddIcon: () => AddIcon,
  AddNoteIcon: () => AddNoteIcon,
  AddPlaylistIcon: () => AddPlaylistIcon,
  AddUserIcon: () => AddUserIcon,
  AirplayIcon: () => AirplayIcon,
  ArrowCollapseIcon: () => ArrowCollapseIcon,
  ArrowCollapseInIcon: () => ArrowCollapseInIcon,
  ArrowExpandIcon: () => ArrowExpandIcon,
  ArrowExpandOutIcon: () => ArrowExpandOutIcon,
  ArrowLeftIcon: () => ArrowLeftIcon,
  BookmarkIcon: () => BookmarkIcon,
  CameraIcon: () => CameraIcon,
  ChaptersIcon: () => ChaptersIcon,
  ChatCollapseIcon: () => ChatCollapseIcon,
  ChatIcon: () => ChatIcon,
  ChevronDownIcon: () => ChevronDownIcon,
  ChevronRightIcon: () => ChevronRightIcon,
  ChromecastIcon: () => ChromecastIcon,
  ClipIcon: () => ClipIcon,
  ClosedCaptionsIcon: () => ClosedCaptionsIcon,
  ClosedCaptionsOnIcon: () => ClosedCaptionsOnIcon,
  CommentIcon: () => CommentIcon,
  ComputerIcon: () => ComputerIcon,
  DeviceIcon: () => DeviceIcon,
  DownloadIcon: () => DownloadIcon,
  EpisodesIcon: () => EpisodesIcon,
  EyeIcon: () => EyeIcon,
  FastBackwardIcon: () => FastBackwardIcon,
  FastForwardIcon: () => FastForwardIcon,
  FlagIcon: () => FlagIcon,
  FullscreenArrowExitIcon: () => FullscreenArrowExitIcon,
  FullscreenArrowIcon: () => FullscreenArrowIcon,
  FullscreenExitIcon: () => FullscreenExitIcon,
  FullscreenIcon: () => FullscreenIcon,
  HeartIcon: () => HeartIcon,
  Icon: () => Icon,
  InfoIcon: () => InfoIcon,
  LanguageIcon: () => LanguageIcon,
  LinkIcon: () => LinkIcon,
  LockClosedIcon: () => LockClosedIcon,
  LockOpenIcon: () => LockOpenIcon,
  MediaFullscreenButton: () => MediaFullscreenButton,
  MediaLiveIndicator: () => MediaLiveIndicator,
  MediaMuteButton: () => MediaMuteButton,
  MediaOutlet: () => MediaOutlet,
  MediaPlayButton: () => MediaPlayButton,
  MediaPlayer: () => MediaPlayer,
  MediaPoster: () => MediaPoster,
  MediaSeekButton: () => MediaSeekButton,
  MediaSlider: () => MediaSlider,
  MediaSliderThumbnail: () => MediaSliderThumbnail,
  MediaSliderValue: () => MediaSliderValue,
  MediaSliderVideo: () => MediaSliderVideo,
  MediaTime: () => MediaTime,
  MediaTimeSlider: () => MediaTimeSlider,
  MediaToggleButton: () => MediaToggleButton,
  MediaVolumeSlider: () => MediaVolumeSlider,
  MenuHorizontalIcon: () => MenuHorizontalIcon,
  MenuVerticalIcon: () => MenuVerticalIcon,
  MicrophoneIcon: () => MicrophoneIcon,
  MobileIcon: () => MobileIcon,
  MoonIcon: () => MoonIcon,
  MusicIcon: () => MusicIcon,
  MusicOffIcon: () => MusicOffIcon,
  MuteIcon: () => MuteIcon,
  NextIcon: () => NextIcon,
  NotificationIcon: () => NotificationIcon,
  OdometerIcon: () => OdometerIcon,
  PauseIcon: () => PauseIcon,
  PictureInPictureExitIcon: () => PictureInPictureExitIcon,
  PictureInPictureIcon: () => PictureInPictureIcon,
  PlayIcon: () => PlayIcon,
  PlaybackSpeedCircleIcon: () => PlaybackSpeedCircleIcon,
  PlaylistIcon: () => PlaylistIcon,
  PreviousIcon: () => PreviousIcon,
  QuestionMarkIcon: () => QuestionMarkIcon,
  QueueListIcon: () => QueueListIcon,
  RadioButtonIcon: () => RadioButtonIcon,
  RadioButtonSelectedIcon: () => RadioButtonSelectedIcon,
  RepeatIcon: () => RepeatIcon,
  RepeatOnIcon: () => RepeatOnIcon,
  RepeatSquareIcon: () => RepeatSquareIcon,
  RepeatSquareOnIcon: () => RepeatSquareOnIcon,
  ReplayIcon: () => ReplayIcon,
  RotateIcon: () => RotateIcon,
  SearchIcon: () => SearchIcon,
  SeekBackward10Icon: () => SeekBackward10Icon,
  SeekBackward15Icon: () => SeekBackward15Icon,
  SeekBackward30Icon: () => SeekBackward30Icon,
  SeekBackwardIcon: () => SeekBackwardIcon,
  SeekForward10Icon: () => SeekForward10Icon,
  SeekForward15Icon: () => SeekForward15Icon,
  SeekForward30Icon: () => SeekForward30Icon,
  SeekForwardIcon: () => SeekForwardIcon,
  SendIcon: () => SendIcon,
  SettingsIcon: () => SettingsIcon,
  SettingsMenuIcon: () => SettingsMenuIcon,
  SettingsSwitchIcon: () => SettingsSwitchIcon,
  ShareArrowIcon: () => ShareArrowIcon,
  ShareIcon: () => ShareIcon,
  ShuffleIcon: () => ShuffleIcon,
  ShuffleOnIcon: () => ShuffleOnIcon,
  StopIcon: () => StopIcon,
  SubtitlesIcon: () => SubtitlesIcon,
  SunIcon: () => SunIcon,
  TheatreModeExitIcon: () => TheatreModeExitIcon,
  TheatreModeIcon: () => TheatreModeIcon,
  ThumbsDownIcon: () => ThumbsDownIcon,
  ThumbsUpIcon: () => ThumbsUpIcon,
  TimerIcon: () => TimerIcon,
  TranscriptIcon: () => TranscriptIcon,
  TvIcon: () => TvIcon,
  UserIcon: () => UserIcon,
  VolumeHighIcon: () => VolumeHighIcon,
  VolumeLowIcon: () => VolumeLowIcon,
  XMarkIcon: () => XMarkIcon,
  useMediaPlayer: () => useMediaPlayer,
  useMediaProvider: () => useMediaProvider,
  useMediaRemote: () => useMediaRemote,
  useMediaStore: () => useMediaStore,
  useSliderStore: () => useSliderStore
});
module.exports = __toCommonJS(src_exports);

// src/components.ts
var import_react = require("maverick.js/react");
var import_vidstack = require("vidstack");
var MediaPlayer = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.PlayerDefinition);
var MediaOutlet = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.OutletDefinition);
var MediaFullscreenButton = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.FullscreenButtonDefinition);
var MediaLiveIndicator = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.LiveIndicatorDefinition);
var MediaMuteButton = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.MuteButtonDefinition);
var MediaPlayButton = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.PlayButtonDefinition);
var MediaPoster = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.PosterDefinition);
var MediaSeekButton = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.SeekButtonDefinition);
var MediaSliderThumbnail = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.SliderThumbnailDefinition);
var MediaSliderValue = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.SliderValueDefinition);
var MediaSliderVideo = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.SliderVideoDefinition);
var MediaSlider = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.SliderDefinition);
var MediaTimeSlider = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.TimeSliderDefinition);
var MediaTime = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.TimeDefinition);
var MediaToggleButton = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.ToggleButtonDefinition);
var MediaVolumeSlider = /* @__PURE__ */ (0, import_react.createLiteReactElement)(import_vidstack.VolumeSliderDefinition);

// src/use-media-player.ts
var import_react2 = require("maverick.js/react");
var import_react3 = require("react");
var import_vidstack2 = require("vidstack");
function useMediaPlayer() {
  const [player, setPlayer] = (0, import_react3.useState)(null), context = (0, import_react2.useReactContext)(import_vidstack2.mediaContext);
  if (false) {
    throw Error(
      "[vidstack] no media context was found - was this called outside of `<MediaPlayer`?"
    );
  }
  (0, import_react3.useEffect)(() => {
    if (!context)
      return;
    const player2 = context.$player();
    player2.onAttach(() => void setPlayer(player2));
    return () => setPlayer(null);
  }, []);
  return player;
}

// src/use-media-provider.ts
var import_maverick = require("maverick.js");
var import_react4 = require("maverick.js/react");
var import_react5 = require("react");
var import_vidstack3 = require("vidstack");
function useMediaProvider() {
  const [provider, setProvider] = (0, import_react5.useState)(null), context = (0, import_react4.useReactContext)(import_vidstack3.mediaContext);
  if (false) {
    throw Error(
      "[vidstack] no media context was found - was this called outside of `<MediaPlayer`?"
    );
  }
  (0, import_react5.useEffect)(() => {
    if (!context)
      return;
    return (0, import_maverick.effect)(() => {
      setProvider(context.$provider());
    });
  }, []);
  return provider;
}

// src/use-media-remote.ts
var import_maverick2 = require("maverick.js");
var import_react6 = require("maverick.js/react");
var import_std = require("maverick.js/std");
var import_react7 = require("react");
var import_vidstack4 = require("vidstack");
function useMediaRemote(target) {
  const context = (0, import_react6.useReactContext)(import_vidstack4.mediaContext), remote = (0, import_react7.useMemo)(() => new import_vidstack4.MediaRemoteControl(), []);
  (0, import_react7.useEffect)(() => {
    const player = context && (0, import_maverick2.peek)(context.$player);
    const ref = !(0, import_std.isUndefined)(target) ? target && "current" in target ? target.current : target : player;
    remote.setTarget(ref || null);
    remote.setPlayer(player || null);
  }, [target]);
  return remote;
}

// src/use-media-store.ts
var import_react9 = require("maverick.js/react");
var import_vidstack5 = require("vidstack");

// src/use-store.ts
var import_maverick3 = require("maverick.js");
var import_std2 = require("maverick.js/std");
var import_react8 = require("react");
function useStore(factory, ref, init) {
  const [$store, $setStore] = (0, import_react8.useState)(init), [_, update] = (0, import_react8.useState)(0), tracking = (0, import_react8.useRef)({
    $props: (0, import_maverick3.signal)([]),
    observing: /* @__PURE__ */ new Set()
  });
  (0, import_react8.useEffect)(() => {
    if (ref == null ? void 0 : ref.current)
      $setStore(ref.current.$store);
  }, []);
  (0, import_react8.useEffect)(() => {
    if (!$store)
      return;
    return (0, import_maverick3.effect)(() => {
      const props = tracking.current.$props();
      for (let i = 0; i < props.length; i++)
        $store[props[i]];
      update((n) => n + 1);
    });
  }, [$store]);
  return (0, import_react8.useMemo)(() => {
    if (!$store)
      return factory.initial;
    const { observing, $props } = tracking.current;
    return new Proxy($store, {
      get(_2, prop) {
        if (!observing.has(prop)) {
          $props.set((prev) => [...prev, prop]);
          observing.add(prop);
        }
        return $store[prop];
      },
      set: import_std2.noop
    });
  }, [$store]);
}

// src/use-media-store.ts
function useMediaStore(ref) {
  const context = (0, import_react9.useReactContext)(import_vidstack5.mediaContext);
  if (false) {
    console.warn(
      `[vidstack] \`useMediaStore\` requires \`RefObject<MediaPlayerElement>\` argument if called outside the <MediaPlayer> component`
    );
  }
  return useStore(import_vidstack5.mediaStore, ref, context == null ? void 0 : context.$store);
}

// src/use-slider-store.ts
var import_react10 = require("maverick.js/react");
var import_vidstack6 = require("vidstack");
function useSliderStore(ref) {
  const $store = (0, import_react10.useReactContext)(import_vidstack6.sliderStoreContext);
  if (false) {
    console.warn(
      `[vidstack] \`useSliderStore\` requires \`RefObject<MediaSliderElement>\` argument if called outside a media slider component`
    );
  }
  return useStore(import_vidstack6.sliderStore, ref, $store);
}

// src/icon.ts
var import_react11 = require("react");
var Icon = /* @__PURE__ */ (0, import_react11.forwardRef)((props, ref) => {
  const { width, height, size = 32, __paths, ...restProps } = props;
  return (0, import_react11.createElement)("svg", {
    ...restProps,
    width: width ?? size,
    height: height ?? size,
    viewBox: "0 0 32 32",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    "aria-hidden": "true",
    focusable: "false",
    "data-media-icon": "true",
    ref,
    dangerouslySetInnerHTML: { __html: __paths }
  });
});

// src/icons.ts
var import_media_icons = require("media-icons");
var import_react12 = require("react");
var AddNoteIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.addNotePaths });
});
var AddPlaylistIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.addPlaylistPaths });
});
var AddUserIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.addUserPaths });
});
var AddIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.addPaths });
});
var AirplayIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.airplayPaths });
});
var ArrowCollapseInIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.arrowCollapseInPaths });
});
var ArrowCollapseIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.arrowCollapsePaths });
});
var ArrowExpandOutIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.arrowExpandOutPaths });
});
var ArrowExpandIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.arrowExpandPaths });
});
var ArrowLeftIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.arrowLeftPaths });
});
var BookmarkIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.bookmarkPaths });
});
var CameraIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.cameraPaths });
});
var ChaptersIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.chaptersPaths });
});
var ChatCollapseIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.chatCollapsePaths });
});
var ChatIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.chatPaths });
});
var ChevronDownIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.chevronDownPaths });
});
var ChevronRightIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.chevronRightPaths });
});
var ChromecastIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.chromecastPaths });
});
var ClipIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.clipPaths });
});
var ClosedCaptionsOnIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.closedCaptionsOnPaths });
});
var ClosedCaptionsIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.closedCaptionsPaths });
});
var CommentIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.commentPaths });
});
var ComputerIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.computerPaths });
});
var DeviceIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.devicePaths });
});
var DownloadIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.downloadPaths });
});
var EpisodesIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.episodesPaths });
});
var EyeIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.eyePaths });
});
var FastBackwardIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.fastBackwardPaths });
});
var FastForwardIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.fastForwardPaths });
});
var FlagIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.flagPaths });
});
var FullscreenArrowExitIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.fullscreenArrowExitPaths });
});
var FullscreenArrowIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.fullscreenArrowPaths });
});
var FullscreenExitIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.fullscreenExitPaths });
});
var FullscreenIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.fullscreenPaths });
});
var HeartIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.heartPaths });
});
var InfoIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.infoPaths });
});
var LanguageIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.languagePaths });
});
var LinkIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.linkPaths });
});
var LockClosedIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.lockClosedPaths });
});
var LockOpenIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.lockOpenPaths });
});
var MenuHorizontalIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.menuHorizontalPaths });
});
var MenuVerticalIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.menuVerticalPaths });
});
var MicrophoneIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.microphonePaths });
});
var MobileIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.mobilePaths });
});
var MoonIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.moonPaths });
});
var MusicOffIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.musicOffPaths });
});
var MusicIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.musicPaths });
});
var MuteIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.mutePaths });
});
var NextIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.nextPaths });
});
var NotificationIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.notificationPaths });
});
var OdometerIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.odometerPaths });
});
var PauseIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.pausePaths });
});
var PictureInPictureExitIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.pictureInPictureExitPaths });
});
var PictureInPictureIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.pictureInPicturePaths });
});
var PlayIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.playPaths });
});
var PlaybackSpeedCircleIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.playbackSpeedCirclePaths });
});
var PlaylistIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.playlistPaths });
});
var PreviousIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.previousPaths });
});
var QuestionMarkIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.questionMarkPaths });
});
var QueueListIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.queueListPaths });
});
var RadioButtonSelectedIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.radioButtonSelectedPaths });
});
var RadioButtonIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.radioButtonPaths });
});
var RepeatOnIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.repeatOnPaths });
});
var RepeatSquareOnIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.repeatSquareOnPaths });
});
var RepeatSquareIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.repeatSquarePaths });
});
var RepeatIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.repeatPaths });
});
var ReplayIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.replayPaths });
});
var RotateIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.rotatePaths });
});
var SearchIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.searchPaths });
});
var SeekBackward10Icon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekBackward10Paths });
});
var SeekBackward15Icon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekBackward15Paths });
});
var SeekBackward30Icon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekBackward30Paths });
});
var SeekBackwardIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekBackwardPaths });
});
var SeekForward10Icon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekForward10Paths });
});
var SeekForward15Icon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekForward15Paths });
});
var SeekForward30Icon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekForward30Paths });
});
var SeekForwardIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.seekForwardPaths });
});
var SendIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.sendPaths });
});
var SettingsMenuIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.settingsMenuPaths });
});
var SettingsSwitchIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.settingsSwitchPaths });
});
var SettingsIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.settingsPaths });
});
var ShareArrowIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.shareArrowPaths });
});
var ShareIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.sharePaths });
});
var ShuffleOnIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.shuffleOnPaths });
});
var ShuffleIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.shufflePaths });
});
var StopIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.stopPaths });
});
var SubtitlesIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.subtitlesPaths });
});
var SunIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.sunPaths });
});
var TheatreModeExitIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.theatreModeExitPaths });
});
var TheatreModeIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.theatreModePaths });
});
var ThumbsDownIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.thumbsDownPaths });
});
var ThumbsUpIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.thumbsUpPaths });
});
var TimerIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.timerPaths });
});
var TranscriptIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.transcriptPaths });
});
var TvIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.tvPaths });
});
var UserIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.userPaths });
});
var VolumeHighIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.volumeHighPaths });
});
var VolumeLowIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.volumeLowPaths });
});
var XMarkIcon = /* @__PURE__ */ (0, import_react12.forwardRef)((props, ref) => {
  return (0, import_react12.createElement)(Icon, { ...props, ref, __paths: import_media_icons.xMarkPaths });
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AddIcon,
  AddNoteIcon,
  AddPlaylistIcon,
  AddUserIcon,
  AirplayIcon,
  ArrowCollapseIcon,
  ArrowCollapseInIcon,
  ArrowExpandIcon,
  ArrowExpandOutIcon,
  ArrowLeftIcon,
  BookmarkIcon,
  CameraIcon,
  ChaptersIcon,
  ChatCollapseIcon,
  ChatIcon,
  ChevronDownIcon,
  ChevronRightIcon,
  ChromecastIcon,
  ClipIcon,
  ClosedCaptionsIcon,
  ClosedCaptionsOnIcon,
  CommentIcon,
  ComputerIcon,
  DeviceIcon,
  DownloadIcon,
  EpisodesIcon,
  EyeIcon,
  FastBackwardIcon,
  FastForwardIcon,
  FlagIcon,
  FullscreenArrowExitIcon,
  FullscreenArrowIcon,
  FullscreenExitIcon,
  FullscreenIcon,
  HeartIcon,
  Icon,
  InfoIcon,
  LanguageIcon,
  LinkIcon,
  LockClosedIcon,
  LockOpenIcon,
  MediaFullscreenButton,
  MediaLiveIndicator,
  MediaMuteButton,
  MediaOutlet,
  MediaPlayButton,
  MediaPlayer,
  MediaPoster,
  MediaSeekButton,
  MediaSlider,
  MediaSliderThumbnail,
  MediaSliderValue,
  MediaSliderVideo,
  MediaTime,
  MediaTimeSlider,
  MediaToggleButton,
  MediaVolumeSlider,
  MenuHorizontalIcon,
  MenuVerticalIcon,
  MicrophoneIcon,
  MobileIcon,
  MoonIcon,
  MusicIcon,
  MusicOffIcon,
  MuteIcon,
  NextIcon,
  NotificationIcon,
  OdometerIcon,
  PauseIcon,
  PictureInPictureExitIcon,
  PictureInPictureIcon,
  PlayIcon,
  PlaybackSpeedCircleIcon,
  PlaylistIcon,
  PreviousIcon,
  QuestionMarkIcon,
  QueueListIcon,
  RadioButtonIcon,
  RadioButtonSelectedIcon,
  RepeatIcon,
  RepeatOnIcon,
  RepeatSquareIcon,
  RepeatSquareOnIcon,
  ReplayIcon,
  RotateIcon,
  SearchIcon,
  SeekBackward10Icon,
  SeekBackward15Icon,
  SeekBackward30Icon,
  SeekBackwardIcon,
  SeekForward10Icon,
  SeekForward15Icon,
  SeekForward30Icon,
  SeekForwardIcon,
  SendIcon,
  SettingsIcon,
  SettingsMenuIcon,
  SettingsSwitchIcon,
  ShareArrowIcon,
  ShareIcon,
  ShuffleIcon,
  ShuffleOnIcon,
  StopIcon,
  SubtitlesIcon,
  SunIcon,
  TheatreModeExitIcon,
  TheatreModeIcon,
  ThumbsDownIcon,
  ThumbsUpIcon,
  TimerIcon,
  TranscriptIcon,
  TvIcon,
  UserIcon,
  VolumeHighIcon,
  VolumeLowIcon,
  XMarkIcon,
  useMediaPlayer,
  useMediaProvider,
  useMediaRemote,
  useMediaStore,
  useSliderStore
});
